scrapy_poet/api.py

```diff
@@ -90,5 +110,12 @@ def parse(self, response):
     def parse(*args, page: page_cls, **kwargs):  # type: ignore
         yield page.to_item()  # type: ignore
 
-    setattr(parse, _CALLBACK_FOR_MARKER, True)
-    return parse
+    async def async_parse(*args, page: page_cls, **kwargs):  # type: ignore
+        yield await page.to_item()  # type: ignore
+
+    if iscoroutinefunction(page_cls.to_item):
+        setattr(async_parse, _CALLBACK_FOR_MARKER, True)
+        return async_parse
+    else:
+        setattr(parse, _CALLBACK_FOR_MARKER, True)
+        return parse
```

Comment:
:lipstick: Maybe we should define the methods within the `if/else` that uses them, and even replace `if/else` with an early return approach?

Reply:
I'm not quite sure about this, as having `parse` and `async_parse` in an earlier section, we'd be able to provide some better structured notes about it. For example, this [comment block](https://github.com/scrapinghub/scrapy-poet/pull/66/files#diff-dc2cce7daeaceab39155f854a698ced1d50a1bb16f5cff8af127d8659aeed50aR104-R109).

Regarding the early return approach, that's a good idea. Updated that in https://github.com/scrapinghub/scrapy-poet/pull/66/commits/f1126fb5c1862718e053714ae980672d0420ce8a